pkg/util/querylimits/limiter.go

```diff
@@ -0,0 +1,69 @@
+package querylimits
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/grafana/loki/pkg/util/limiter"
+	logutil "github.com/grafana/loki/pkg/util/log"
+)
+
+type Limiter struct {
+	logger log.Logger
+	limiter.CombinedLimits
+}
+
+func NewLimiter(log log.Logger, original limiter.CombinedLimits) *Limiter {
+	return &Limiter{
+		logger:         log,
+		CombinedLimits: original,
+	}
+}
+
+// MaxQueryLength returns the limit of the length (in time) of a query.
+func (l *Limiter) MaxQueryLength(ctx context.Context, userID string) time.Duration {
+	original := l.CombinedLimits.MaxQueryLength(ctx, userID)
+	requestLimits := ExtractQueryLimitsContext(ctx)
+	if requestLimits == nil || requestLimits.MaxQueryLength == 0 || time.Duration(requestLimits.MaxQueryLength) > original {
+		_ = level.Debug(logutil.WithContext(ctx, l.logger)).Log("msg", "using original limit")
```

Comment:
I don't think that we need to assign to blank identifier.We explicitly ignore that in the linter:

`.golangci.yml`:

```yaml
issues:
  exclude:
    - Error return value of .*log\.Logger\)\.Log\x60 is not checked
    - Error return value of .*.Log.* is not checked
    - Error return value of `` is not checked
```